src/components/Select.jsx

```diff
@@ -75,6 +75,7 @@ const CustomInput = props => {
 
 const CustomOption = props => {
   const ref = useRef();
+  const { dataCy } = props.data;
```

Comment:
```suggestion
  const { dataCy } = props;
```

Why not use dataCy as a prop directly?

Reply:
@VarunSriram99 : I initially tried accessing `dataCy` directly from props, but it didn't work. Later, I found out that react-select places custom data within `props.data`, whereas `label` is a directly accessible prop provided by react-select.